File: _includes/code/indexes/indexes-v3.py
import os
import weaviate

# Instantiate the client with the OpenAI API key
client = weaviate.Client(
    "http://localhost:8080",
    additional_headers={
        "X-OpenAI-Api-Key": os.environ["OPENAI_API_KEY"]
    }
)

# ===============================================
# ===== CREATE A COLLECTION WITH PROPERTIES =====
# ===============================================

# Clean start
if client.schema.exists("Inventory"):
    client.schema.delete_class("Inventory")

# START RangeIndex
class_obj = {
    "class": "Inventory",
    "properties": [
        {
            "name": "itemName",
            "dataType": ["text"],
        },
        {
            "name": "itemDescription",
            "dataType": ["text"],
        },
        {
            "name": "itemCount",
            "dataType": ["int"],
            "indexRangeFilters": True,
        },
    ],
        "invertedIndexConfig": {
        "bm25": {
            "b": 0.7,
            "k1": 1.25
        },
        "indexTimestamps": True,
        "indexNullState": True,
        "indexPropertyLength": True
    }
}

client.schema.create_class(class_obj)  # returns null on success
# END RangeIndex

# Test
schema = client.schema.get("Inventory")
assert schema["properties"][2]['indexRangeFilters'] == True

File: _includes/code/indexes/indexes-v3.ts
// ================================
// ===== INSTANTIATION-COMMON =====
// ================================

import weaviate, { WeaviateClient, vectorIndex } from 'weaviate-client';
import { vectorizer, generative, dataType, tokenization, configure, reconfigure, vectorDistances } from 'weaviate-client';

const client: WeaviateClient = await weaviate.connectToLocal(
 {
   headers: {
     'X-OpenAI-Api-Key': process.env.OPENAI_APIKEY,  // Replace with your inference API key
   }
 }
)

const collectionName = 'Inventory'

// START RangeIndex
let inventory = client.collections.get('Inventory')
// END RangeIndex

// ====================================
// ===== CREATE A CLASS WITH PROPERTIES
// ====================================

// Clean slate
try {
  await client.collections.delete(collectionName)
} catch (e) {
  // ignore error if class doesn't exist
}

// START RangeIndex
import { dataType } from 'weaviate-client';

await client.collections.create({
  name: 'Inventory',
  properties: [
    {
      name: 'itemName',
      dataType: dataType.TEXT,
    },
    {
      name: 'itemDescription',
      dataType: dataType.TEXT,
    },
    {
     name: 'itemCount',
     dataType: dataType.INT,
     indexSomeWeirdNonense: true
   },
  ],
})
// END RangeIndex

const collectionConfig = await inventory.config.get()
console.log(collectionConfig)
